blosc/blosc2.h

```diff
@@ -736,26 +726,32 @@ BLOSC_EXPORT const char* blosc_cbuffer_complib(const void* cbuffer);
 *********************************************************************/
 
 typedef void*   (*blosc2_open_cb)(const char *urlpath, const char *mode, void *params);
-typedef int     (*blosc2_close_cb)(void *stream, void* params);
-typedef int     (*blosc2_seek_cb)(void *stream, int64_t offset, int whence, void* params);
-typedef int64_t (*blosc2_write_cb)(const void *ptr, int64_t size, int64_t nitems, void *stream, void *params);
-typedef int64_t (*blosc2_read_cb)(void *ptr, int64_t size, int64_t nitems, void *stream, void *params);
+typedef int     (*blosc2_close_cb)(void *stream);
+typedef int64_t (*blosc2_tell_cb)(void *stream);
+typedef int     (*blosc2_seek_cb)(void *stream, int64_t offset, int whence);
+typedef int64_t (*blosc2_write_cb)(const void *ptr, int64_t size, int64_t nitems, void *stream);
+typedef int64_t (*blosc2_read_cb)(void *ptr, int64_t size, int64_t nitems, void *stream);
+typedef int64_t (*blosc2_truncate_cb)(void *stream, int64_t size);
 
 typedef struct {
   blosc2_open_cb open;
   blosc2_close_cb close;
+  blosc2_tell_cb tell;
   blosc2_seek_cb seek;
   blosc2_write_cb write;
   blosc2_read_cb read;
+  blosc2_truncate_cb truncate;
   void *params;
```

Comment:
Looks you removed `params` from all the arguments which is good. Also probably need to remove `params` in i/o interface the struct too. Having `params` in open callback is ok.

Reply:
The `params` in the `blosc_io` structure are needed to allow the users to set their params. See an example of use: https://github.com/Blosc/c-blosc2/blob/86eaeba547eb5e15536e8dda7135957191fb1e3b/tests/test_udio.c

Another option may be to split the `blosc_io` structure in two: `blosc_io` (only callbacks) and `blosc_io_params`?

Reply:
Sorry for the delay. It is a unusual design pattern for the `io_params` in `test_udio.c`. I was thinking you can return `io_params` with an accessor method like:
```
struct test_udio_params* test_get_udio_params(void *stream) {
   test_file *my = (test_file *) stream;
   return my->params;
}
```
Then you'd expose the stream handle from `blosc2_schunk` or something like that. But since the stream can get created and deleted multiple times it doesn't make much sense.

I think you are probably better off creating a global `test_udio_params` variable in your `test_udio.c` and removing `void *params` from the `open` signature and from the interface.

Reply:
We have changed a bit the way to register the plugins (so we don't need to pass the callbacks to the storage).  We are going to merge this as-is.  In case you still have a better suggestion, please feel free to send it as a PR.